thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs

```diff
@@ -612,17 +614,18 @@ a3RenderUser uid _ = externalUrlOfDefaultPersona uid
 -- | Convert a local file name into a absolute path relative to the A3 backend endpoint.  (Returns
 -- exposed url.)
 a3backendPath :: ThentosConfig -> ST -> Path
-a3backendPath config localPath = Path $ cs (exposeUrl beHttp) <//> localPath
+a3backendPath config localPath = Path $ a3Prefix <//> localPath
   where
-    beHttp     = case config >>. (Proxy :: Proxy '["backend"]) of
-                     Nothing -> error "a3backendPath: backend not configured!"
-                     Just v -> Tagged v
+    -- FIXME: get rid of the fromJust, a3-Prefix should not be optional
+    a3Prefix = fromJust $ config >>. (Proxy :: Proxy '["a3-prefix"])
 
 userIdFromPath :: MonadError (ThentosError e) m => Path -> m UserId
 userIdFromPath (Path s) = do
     uri <- either (const . throwError . MalformedUserPath $ s) return $
         URI.parseURI URI.laxURIParserOptions $ cs s
     rawId <- maybe (throwError $ MalformedUserPath s) return $
+        -- FIXME: I don't think this will work, as the path may be something
+        --  like /foo/principals/users/ (see https://github.com/liqd/adhocracy3/pull/1849)
```

Comment:
Yes, I think this should be fixed soon, but it's out of scope for this PR. I'll mark it as BUG and open a ticket.